include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -698,45 +719,112 @@ class graph_oriented_layout_design_impl
         {
             timeout = 10000u;
         }
-        bool timeout_limit_reached = false;
 
-        // main A* loop
+        // main loop
         while (!timeout_limit_reached)
         {
-            for (auto& ssg : ssg_vec)
+            // if multithreading is enabled
+            if (ps.enable_multithreading)
             {
-                if (ssg.frontier_flag)
+                // mutex to protect shared resources (if necessary)
+                std::mutex                                   mtx;
```

Comment:
```suggestion
                std::mutex                                   mtx{};
```
Maybe we can rename it to make it clear what it is used for?